Core/src/org/sleuthkit/autopsy/textextractors/TextFileExtractor.java

```diff
@@ -73,7 +76,26 @@ private Reader getReader(Charset encoding) {
     }
 
     @Override
-    public boolean isSupported() {
-        return file.getMIMEType().equals("text/plain");
+    public boolean isSupported() {        
+        // get the MIME type
+        String mimeType = file.getMIMEType();
+        
+        // if it is not present, attempt to use the FileTypeDetector to determine
+        if (StringUtils.isEmpty(mimeType)) {
+            FileTypeDetector fileTypeDetector = null;
+            try {
+                fileTypeDetector = new FileTypeDetector();
+            } catch (FileTypeDetector.FileTypeDetectorInitException ex) {
+                logger.log(Level.SEVERE, "Unable to create file type detector for determining MIME type.");
+                return false;
+            }
+            mimeType = fileTypeDetector.getMIMEType(file);
+            
+            // if able to determine mime type, 
+            if (!StringUtils.isEmpty(mimeType))
+                file.setMIMEType(mimeType);
+        }
```

Comment:
@eugene7646 and @gdicristofaro...

Do you gentlemen realize that AbstractFile.setMIMEType only stores the result in the object, not in the case database? I suspect that is not what was intended here. I'm no longer sure what our policy is: do we save or throw away such calculations? We should check what we do elsewhere, and if the intent is to save here, call AbstractFile.save here.

Reply:
> Do you gentlemen realize that AbstractFile.setMIMEType only stores the result in the object, not in the case database?
The answer is "no". Personally I think we should save it in the database but I agree, we should check what we do in other places.

Reply:
From what I can see, typically the result of FileTypeDetector.getMIMEType() is not persisted like FileViewer.isPreferred().  I don't believe I saw an example that the MIME type is set on a purely in-memory object either.  So, I can keep with this precedent if you would like and remove the file.setMIMEType all together or call save().  It appears that the save() method throws an exception on sql error, which may increase the complexity of the isSupported() method.

Reply:
Let's just remove the call, thanks.

Reply:
Got it.